### PR TITLE
Reimplement AI PoC from GUI1

### LINKED ISSUE #9313: Add back AI POC support as it was in GUI1
- This is a first step in restoring AI functionality.
- Internal feature only. 
- Enabled by adding environment variable with OpenAI key.

Activated by typing AI the component browser followed by the AI prompt.
On pressing enter the CB stays open and the code is replaced with the response from the AI model.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #9313 

[Screencast from 2024-03-22 09-09-07.webm](https://github.com/enso-org/enso/assets/3919101/6ad86145-6882-4bde-993d-b1270f1ec06c)

### Important Notes

* This is PoC, so I didn't spend time on polishing the visuals; the design will likely change.
* I modified the shortcut handler a bit, allowing making multiple actions for same binding - the action's handler will be called in unspecified order, until one of them handle the event (i.e. not return false). 
* To make it working regardless of imports, I needed to export AI module in Standard.Visualization. Moreover, needed to remove build_ai_prompt for Any, because it was causing issues - expect a bug report soon.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by Frizi — APPROVED** [review]

File: distribution/lib/Standard/Visualization/0.0.0-dev/src/AI.enso [hunk 1]
```diff
@@ -28,8 +28,9 @@ Table.build_ai_prompt self =
     goal_line = "goal=" + goal_placeholder + "==>>`"
     base_prompt + '\n' + columns + goal_line
 
-## PRIVATE
-Any.build_ai_prompt self = "````"
+# TODO [ao]: commented, because it caused weird responses in GUI - see https://github.com/enso-org/enso/issues/9518
+# ## PRIVATE
+# Any.build_ai_prompt self = "````"
```
Comment (radeusgd) [inline distribution/lib/Standard/Visualization/0.0.0-dev/src/AI.enso:33]:
```suggestion
Any.build_ai_prompt self =
	"goal=" + goal_placeholder + "==>>`"
```

Wouldn't this fix it?

Comment (radeusgd) [reply distribution/lib/Standard/Visualization/0.0.0-dev/src/AI.enso:33]: Ah, looking at #9518 probably not.

**Review by radeusgd — COMMENTED** [review]
What happens if I type `AI:` on a node that _is not_ a Table?

**Comment by farmaazon** [comment]
> What happens if I type `AI:` on a node that _is not_ a Table?

We show the error, that there's no build_ai_prompt for <type>. But I guess it will be fixed soon in #9518

**Review by radeusgd — APPROVED** [review]
Approving the `.enso` file changes.

We probably need a way to work with Any, but I see it's tracked by follow up task, so for this PoC it seems all OK.

**Review by 4e6 — APPROVED** [review]

**Pull request merged**